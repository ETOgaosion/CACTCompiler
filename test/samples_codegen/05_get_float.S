iobuff:
	.zero 80
.text
.align 1
.globl	main
.type	main, @function
main:
	sd	ra, -8(sp)
	fmv.w.x	ft0, zero
	fsw	ft0, -12(sp)
	addi	sp, sp, -16
	call	get_float
	addi	sp, sp, 16
	fsw	fa0, -16(sp)
	lw	t0, -16(sp)
	sw	t0, -12(sp)
	flw	fa0, -12(sp)
	addi	sp, sp, -16
	call	print_float
	addi	sp, sp, 16
	mv	a0, zero
	ld	ra, -8(sp)
	ret
