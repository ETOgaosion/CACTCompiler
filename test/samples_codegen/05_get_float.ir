============ global var ================
============ imm value ================
============ functions ================
----------- funcDef -----------
main() => 2:
----------- paramVars -----------
----------- localVars -----------
a := symbol type: 1; data type: 3; initValue: 0

----------- tempVars ------------
t_0:= symbol type: TEMP VAR; data type: 3
----------- Codes ------------
main:
	a = 0;
	call get_float; 
	t_0 =  return ;
	t_1 <= a;
	t_1 = t_0;
	addparam t_1;
	call print_float; 
	return 0;

