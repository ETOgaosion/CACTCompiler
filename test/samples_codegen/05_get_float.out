bbl loader
please enter an float number:
2.400000
